test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -197,6 +199,82 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBinIntegrationTest do
       )
     end
 
+    @tag :tmp_dir
+    test "test creation of delta playlist", %{tmp_dir: tmp_dir} do
+      alias Membrane.HTTPAdaptiveStream.Storages.FileStorage
+
+      hackney_sources =
+        @audio_multiple_video_tracks_sources
+        |> Enum.map(fn {path, encoding, name} ->
+          {%Membrane.Hackney.Source{location: path, hackney_opts: [follow_redirect: true]},
+           encoding, name}
+        end)
+
+      pipeline =
+        Testing.Pipeline.start_link_supervised!(
+          module: TestPipeline,
+          custom_args: %{
+            sources: hackney_sources,
+            hls_mode: @pipeline_config.hls_mode,
+            partial_segments: true,
+            target_window_duration: @pipeline_config.target_window_duration,
+            persist?: @pipeline_config.persist?,
+            storage: %FileStorage{
+              directory: tmp_dir
+            }
+          }
+        )
+
+      assert_pipeline_play(pipeline)
+      assert_pipeline_notified(pipeline, :sink_bin, :end_of_stream, 10_000)
+
+      # Give some time to save all of the files to disk
+      Process.sleep(10_000)
```

Comment:
I don't think we need to additionally wait for the files to be safe, `end_of_stream` ensures us that everything is already saved.
```suggestion
```
You can also remove sleep() in two other cases.